lib/middleware/account_middleware.rb

```diff
@@ -0,0 +1,36 @@
+## Multitenant Account Middleware
+#
+# Included in the Rails engine if enabled.
+#
+# Used for setting the Account by the first ID in the URL like Basecamp 3.
+# This means we don't have to include the Account ID in every URL helper.
+# Cloned from JumpstartRails
+
+class AccountMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  # http://example.com/12345/projects
+  def call(env)
+    request = ActionDispatch::Request.new env
+    _, account_id, request_path = request.path.split("/", 3)
+
+    puts ['->', _, account_id, request_path].inspect
+
+    if /\d+/.match?(account_id)
+      # We'll use the User model instead of an Account model
+      # In real life we'd use the Account model or wtv defines our tenancy
+      if (account = User.find_by(id: account_id))
+        Current.account = account
```

Comment:
can the account be stored via [context](https://docs.avohq.io/1.0/customization.html#context)? 

Something like:

```ruby
Avo.configure do |config|
  config.set_context do
    {
      account: User.find_by(id: request.params[:tenant_id]) # or whatever resource defines a tenancy in routes
    }
  end
end
```